### PR TITLE
add filters tech level and tech base

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
- add filter tech level 
- add filter tech base

image

### CODE REVIEW

**Review by HammerGS — APPROVED** [review]
Testing this and works fine.

**Comment by SJuliez** [comment]
Just curious: Do these two fields lead to different results from the fields we have in the unit selector main window?

**Comment by kuronekochomusuke** [comment]
yes, it is different.   The new Tech Level and Tech Base come from the values set in MegaMekLab and are shown on the summary tab of the mech preview in the add a combat unit search.  

Not sure where the Type: in the add a combat unit search comes from.

**Comment by kuronekochomusuke** [comment]
would think the tech level should be the same, but they are not.  

could be these are old files that need to be updated, to have the tech level set correctly.

Type: unofficial
image

Tech Level: unofficial
image

File: megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java [hunk 1]
```diff
@@ -445,6 +451,8 @@ private JPanel createBasePanel() {
         loadTriStateItem(Mech.COCKPIT_STRING, listCockpitType, 7);
         loadTriStateItem(EquipmentType.structureNames, listInternalsType, 7);
         loadTriStateItem(Engine.getEngineTypes(), listEngineType, 5);
+        loadTriStateItem(SimpleTechLevel.getDescriptions(), listTechLevel, 5);
```
Comment (SJuliez) [inline megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java:454]: I have noticed in a previous PR but didn't comment on it. Would you mind adding a version of this that ccepts a List<String>. Lists are generally preferable to arrays for safety and general ease of handling and well, modernity. Generating that list im SimpleTechLevel is at least one method call shorter then. The getEngineTypes method would also be simpler.

**Review by SJuliez — CHANGES_REQUESTED** [review]

File: megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java:454]: a `List<String>`
the formatting swallowed that :)

**Review by SJuliez — APPROVED** [review]

**Pull request merged**